controllers/volume/v1alpha2/volume_controller_unit_test.go

```diff
@@ -570,6 +570,82 @@ func unitTestsReconcile() {
 			})
 		})
 
+		When("VM Spec.Volumes has CNS volume with an existing CnsNodeVmAttachment for a different VM", func() {
+
+			When("CnsNodeVmAttachment owned by different VM", func() {
+
+				BeforeEach(func() {
+					vmVol = *vmVolumeWithPVC1
+					vm.Spec.Volumes = append(vm.Spec.Volumes, vmVol)
+
+					attachment := cnsAttachmentForVMVolume(vm, vmVol)
+					attachment.ObjectMeta.OwnerReferences[0].UID = "some-other-uuid"
+					attachment.Spec.NodeUUID = "some-other-bios-uuid"
+					attachment.Status.Attached = true
+					attachment.Status.AttachmentMetadata = map[string]string{
+						volume.AttributeFirstClassDiskUUID: dummyDiskUUID,
+					}
+					initObjects = append(initObjects, attachment)
+				})
+
+				It("returns expected error", func() {
+					err := reconciler.ReconcileNormal(volCtx)
+					Expect(err).To(HaveOccurred())
+					Expect(err.Error()).To(ContainSubstring(" has a different controlling owner"))
+
+					By("Expected VM Status.Volumes", func() {
+						Expect(vm.Status.Volumes).To(BeEmpty())
+					})
+
+					By("attachment still exists", func() {
+						Expect(getCNSAttachmentForVolumeName(vm, vmVol.Name)).ToNot(BeNil())
+					})
+				})
+			})
+
+			When("CnsNodeVmAttachment owned by different VM", func() {
```

Comment:
can we change this to maybe `owned by a VM with a different BiosUUID` to different from the previous test that's more about the ownerRef being different?

Reply:
Oops copy-pasta here. I'll fix it. Thanks.